bokehjs/src/lib/models/glyphs/webgl/base.ts

```diff
@@ -32,35 +31,13 @@ export abstract class BaseGLGlyph {
 
   render(_ctx: Context2d, indices: number[], mainglyph: GlyphView): boolean {
     if (indices.length == 0) {
-      // Implementations assume at least one index to draw. We return true,
-      // because there is no need to switch back to a fallback renderer.
       return true
     }
-    // Get transform
-    const [a, b, c] = [0, 1, 2]
-    let wx = 1   // Weights to scale our vectors
-    let wy = 1
-    let [dx, dy] = this.glyph.renderer.coordinates.map_to_screen([a*wx, b*wx, c*wx], [a*wy, b*wy, c*wy])
-    if (isNaN(dx[0] + dx[1] + dx[2] + dy[0] + dy[1] + dy[2])) {
-      logger.warn(`WebGL backend (${this.glyph.model.type}): falling back to canvas rendering`)
-      return false
-    }
-    // Try again, but with weighs so we're looking at ~100 in screen coordinates
-    wx = 100 / Math.min(Math.max(Math.abs(dx[1] - dx[0]), 1e-12), 1e12)
-    wy = 100 / Math.min(Math.max(Math.abs(dy[1] - dy[0]), 1e-12), 1e12)
-    ;[dx, dy] = this.glyph.renderer.coordinates.map_to_screen([a*wx, b*wx, c*wx], [a*wy, b*wy, c*wy])
-    // Test how linear it is
-    if ((Math.abs((dx[1] - dx[0]) - (dx[2] - dx[1])) > 1e-4) ||
-        (Math.abs((dy[1] - dy[0]) - (dy[2] - dy[1])) > 1e-4)) {
-      logger.warn(`WebGL backend (${this.glyph.model.type}): falling back to canvas rendering`)
-      return false
-    }
-    const [sx, sy] = [(dx[1]-dx[0]) / wx, (dy[1]-dy[0]) / wy]
```

Comment:
Does this mean we can use non-linear scales already at this point? If so, a test would be useful to see a test to confirm.

Reply:
Yes this works tho at least in this particular example interactive panning is a bit laggy 

<img width="878" alt="Screen Shot 2020-08-12 at 8 32 17 AM" src="https://user-images.githubusercontent.com/1078448/90035206-66f7f800-dc76-11ea-978e-4691d9f82224.png">

I'll add a test though before merging

Reply:
test added. It does not verify that webgl is used but then again we don't have any fallback conditions anymore